lib/encryption/utils/bootstrap.dart

```diff
@@ -479,33 +482,20 @@ class Bootstrap {
               ));
       Logs().v('Device signing keys have been uploaded.');
       // aaaand set the SSSS secrets
-      final futures = <Future<void>>[];
       if (masterKey != null) {
-        futures.add(
-          client.onSync.stream
-              .firstWhere((syncUpdate) =>
-                  masterKey?.publicKey != null &&
-                  client.userDeviceKeys[client.userID]?.masterKey?.ed25519Key ==
-                      masterKey?.publicKey)
-              .then((_) => Logs().v('New Master Key was created')),
-        );
+        while (!(masterKey.publicKey != null &&
+            client.userDeviceKeys[client.userID]?.masterKey?.ed25519Key ==
+                masterKey.publicKey)) {
+          Logs().v('Waiting for master to be created');
+          await client.oneShotSync(timeout: Duration(seconds: 3));
+        }
       }
       for (final entry in secretsToStore.entries) {
-        futures.add(
-          client.onSync.stream
-              .firstWhere((syncUpdate) =>
-                  syncUpdate.accountData != null &&
-                  syncUpdate.accountData!
-                      .any((accountData) => accountData.type == entry.key))
-              .then((_) =>
-                  Logs().v('New Key with type ${entry.key} was created')),
-        );
         Logs().v('Store new SSSS key ${entry.key}...');
         await newSsssKey?.store(entry.key, entry.value);
```

Comment:
You now made the await sequential, why not still do it in parallel using Future.wait?

Reply:
ah I assumed doing it sequentially on `store` would cause all kinds of mishaps with db stuff. Previously the db was sequential but waiting for accountdata was parallel.

Reply:
Imo you should still store all the secrets in parallel, so that you don't need to wait for 3 syncs sequentially.